BigQuery/tests/Snippet/ModelTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\BigQuery\Tests\Snippet;
+
+use Google\Cloud\BigQuery\Connection\ConnectionInterface;
+use Google\Cloud\BigQuery\Model;
+use Google\Cloud\Core\Testing\Snippet\SnippetTestCase;
+use Prophecy\Argument;
+
+/**
+* @group bigquery
+*/
+class ModelTest extends SnippetTestCase
+{
+    private $connection;
+    const PROJECT_ID = 'myProjectId';
+    const DATASET_ID = 'myDatasetId';
+    const MODEL_ID = 'myModelId';
+
+    public function setUp()
+    {
+        $this->connection = $this->prophesize(ConnectionInterface::class);
+    }
+
+    public function getModel($connection, array $info = [])
+    {
+        return new Model(
+            $connection->reveal(),
+            self::MODEL_ID,
+            self::DATASET_ID,
+            self::PROJECT_ID,
+            $info
```

Comment:
Assuming my proposal goes through this is fine to keep as is, but currently `Model` does not accept this argument.